semgrep/semgrep/config_resolver.py

```diff
@@ -455,19 +466,34 @@ def saved_snippet_to_url(snippet_id: str) -> str:
     return registry_id_to_url(f"s/{snippet_id}")
 
 
-def resolve_config(config_str: str) -> Dict[str, YamlTree]:
-    """ resolves if config arg is a registry entry, a url, or a file, folder, or loads from defaults if None"""
-    start_t = time.time()
+def registry_url(config_str: str) -> Optional[str]:
+    """ returns the registry url for the config_str, if it is a registry entry """
     if config_str in RULES_REGISTRY:
-        config = download_config(RULES_REGISTRY[config_str])
+        metric_manager.using_server = True
+        return RULES_REGISTRY[config_str]
     elif is_url(config_str):
-        config = download_config(config_str)
+        metric_manager.using_server = True
+        return config_str
     elif is_registry_id(config_str):
-        config = download_config(registry_id_to_url(config_str))
+        metric_manager.using_server = True
+        return registry_id_to_url(config_str)
     elif is_saved_snippet(config_str):
-        config = download_config(saved_snippet_to_url(config_str))
+        metric_manager.using_server = True
+        return saved_snippet_to_url(config_str)
+    else:
+        return None
+
+
+def resolve_config(config_str: str) -> Mapping[str, YamlTree]:
+    """ resolves if config arg is a registry entry, a url, or a file, folder, or loads from defaults if None"""
+    start_t = time.time()
+
+    config_url = registry_url(config_str)
```

Comment:
This is a little unsafe, TODO will implement this better first thing if text output is approved